weio/hawc2_dat_file.py

```diff
@@ -16,28 +16,65 @@ def defaultExtensions():
     def formatName():
         return 'HAWC2 dat file'
 
-    def __init__(self, filename=None, **kwargs):
+    def __init__(self, filename=None, streaming=False, **kwargs):
         self.info={}
-        self.data=np.array([])
+        self.data=None if streaming else np.array([])
         self.bHawc=False
-        super(HAWC2DatFile, self).__init__(filename=filename,**kwargs)
+        self._res_file = None  # Store ReadHawc2 object for streaming
+
+        # Call parent __init__ but handle streaming case
+        File.__init__(self)

```

Comment:
Streaming and filename should be handled by the parent: 

`File.__init__(self, filename=filename, streaming=streaming)`